### ISSUE TITLE
Cannot decrypt data

### ISSUE THREAD

**Issue by joshsmoore** [krzyzanowskim/ObjectivePGP#16]
Hi. I believe this is a newbie question (I am new to both ObjectiveC and PGP).  Right now I am trying to just encrypt and decrypt data.  The encryption goes with no problems.  However, when I try to decrypt the data I get an error: `Error Domain=ObjectivePGP Code=0 "Valid PublicKeyEncryptedSessionKeyPacket not found" UserInfo=0x7fdd7041d430 {NSLocalizedDescription=Valid PublicKeyEncryptedSessionKeyPacket not found}`

I believe my code matches the example given in the ready me but I will post it here:

``` objectivec
ObjectivePGP *pgp = [[ObjectivePGP alloc] init];

    NSArray *keys = [pgp importKeysFromFile:[[[NSBundle mainBundle] resourcePath] stringByAppendingString:@"/850ED5AE.asc"] allowDuplicates:NO];

    PGPKey *keyToEncrypt = [keys objectAtIndex:0];

    /* encrypt data, armor output (ASCII file)  */
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *documentsDirectory = [paths objectAtIndex:0];


    NSError *error = nil;
    NSData *encryptedData = [pgp encryptData:[@"this is encrypted" dataUsingEncoding:NSUTF8StringEncoding] usingPublicKey:keyToEncrypt armored:YES error:&error];
    NSLog(@"got passed encrypt call");
    if (encryptedData && !error) {
        NSLog(@"encryption success");
        [encryptedData writeToFile:[documentsDirectory stringByAppendingPathComponent:@"/encrypted_file.txt.gpg"] atomically:YES];
    }

    error = nil;

    NSData *decryptedData = [pgp decryptData:encryptedData passphrase:nil error:&error];
    if (decryptedData && !error) {
        NSLog(@"decryption success");
    } else {
        NSLog(@"%@", error);
    }
```

Thanks for he help and the library!

Josh

**Comment by jlalvarez18** [comment]
hey @joshsmoore if im not mistaken, you must import the private key in order to decrypt the data.

@krzyzanowskim would you be able to add a method where you can pass in the private key when decrypting the data?

**Comment by joshsmoore** [comment]
@jlalvarez18 thanks that did the trick!  my problem was I had the public and private key in the same file and I expected "importKeysFromFile" to import them all (since it was plural). But, when I split them into separate files it worked fine.  This maybe a stupid question but if it returns an Array and the method name is plural why only read 1 key per file?

Thanks again.

**Comment by jlalvarez18** [comment]
@joshsmoore yeah I thought the exact same thing :)

**Comment by krzyzanowskim** [comment]
It's fixed in 0.6. Feel free to reopen this issue if you encounter this issue with the current release.
